src/cls/rbd/cls_rbd.cc

```diff
@@ -5046,14 +5046,14 @@ int trash_list(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
 
   map<string, cls::rbd::TrashImageSpec> data;
   string last_read = trash::image_key(start_after);
-  int max_read = RBD_MAX_KEYS_READ;
+  int max_read;
   bool more = true;
 
   CLS_LOG(20, "trash_get_images");
   while (data.size() < max_return) {
     map<string, bufferlist> raw_data;
-    int max_read = std::min<int32_t>(RBD_MAX_KEYS_READ,
-                                     max_return - data.size());
+    max_read = std::min<int32_t>(RBD_MAX_KEYS_READ,
```

Comment:
Keep the variable scoping as tight as possible -- keep the original line here and just drop line 5049 since it was being shadowed.